lib/facterdb.rb

```diff
@@ -2,11 +2,23 @@
 require 'jgrep'
 
 module FacterDB
+  # returns Hash of filenames to fact Hashes
+  def self.build_database( fs )
+    Hash[
+      Dir.glob(fs).map do |f|
+        t = File.read(f)
+        [f,JSON.parse(t)]
+      end
+    ]
+  end
+
 
   def self.database
-    @database ||= "[#{Dir.glob("#{File.expand_path(File.join(File.dirname(__FILE__), '../facts'))}/*/*.facts").map { |f| File.read(f) }.join(',')}]\n"
+    fs = File.expand_path('../facts/*/*.facts',File.dirname(__FILE__))
```

Comment:
It does a few things, but the intent was to be more semantic and less error-prone when a file contains bad data (or a different encoding).  

A convenient side effect is that `self.buld_database()` makes a nice place to insert code for inspecting incoming data.  I've already done this to identify broken data files during development.   In the future, we might consider using it to identify facter data that doesn't conform to FacterDB conventions (e.g., an `fqdn` of `foo.example.com`)